keps/sig-storage/3141-prevent-volume-mode-conversion/README.md

```diff
@@ -0,0 +1,724 @@
+
+#### Alpha -> Beta
+
+- One release with positive feedback from users. 
+
+#### Beta -> GA
+
+- Deployed in production in use by backup vendors. 
+- Gone through one kubernetes upgrade.
+
+### Upgrade / Downgrade Strategy
+
+1. Upgrading `external-snapshotter` and `external-provisioner` with `PreventVolumeModeConversion`
+enabled:
+- `VolumeSnapshots` created after the upgrade will maintain a reference to the
+source volume mode. Newly created `PVCs` will undergo an additional check before the 
+provisioning is performed on the storage backend. 
+- `VolumeSnapshots` created before the upgrade will leave the new API field 
+unpopulated.
+
+2. Downgrading `external-snapshotter` and `external-provisioner` with `PreventVolumeModeConversion`
+disabled: 
+- `VolumeSnapshots` created priot to the upgrade will still maintain a reference 
+to the source volume mode, but `PVCs` can be created from them without the
+additional check. 
+
+### Version Skew Strategy
+
+This proposal requires changes to three components - `VolumeSnapshotContent` API, 
+`external-snapshotter` and `external-provisioner`. 
+
+If any of the components are not upgraded to a version supporting this feature,
+then the feature will not work as expected. From an end user perspective, the
+existing behavior will continue, ie, there will be no check to prevent 
+unauthorized conversion of the volume mode. 
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Other
+  - Describe the mechanism: Out-of-tree flag named `PreventVolumeModeConversion`, 
+  which will be enabled in `external-provisioner` and `external-snapshotter`.
+  - Will enabling / disabling the feature require downtime of the control
+    plane? No
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node? (Do not assume `Dynamic Kubelet Config` feature is enabled).
+    No
+
+###### Does enabling the feature change any default behavior?
+
+Yes. Users without requisite privileges cannot alter the volume mode of `VolumeSnapshot`
+when it is being used to create a `PVC`. Users with privileges need to add an 
+annotation to the corresponding `VolumeSnapshotContent` instance if they
+require the volume mode to be converted.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes. Disabling the feature is supported and will fall back to the existing behavior.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+The new behaviour will be re enabled. `VolumeSnapshots` created when the feature
+was disabled will not have the new capabilities.
+
+###### Are there any tests for feature enablement/disablement?
+
+We will add unit tests with and without the feature gate enabled. The expectation
+is for new fields in `VolumeSnapshotContent` to be dropped when the feature gate
```

Comment:
I'm not sure if we should "drop" the new field as we don't have a real feature gate. Maybe we should just ignore the new field if the feature flag is disabled. You can keep this as is. I'll let others comment on this.